operators/src/processing/meteosat/radiance.rs

```diff
@@ -0,0 +1,502 @@
+            Grid2D::new(
+                [3, 2].into(),
+                vec![13.0, 15.0, 17.0, 19.0, 21.0, no_data_value_option.unwrap()],
+                no_data_value_option,
+            )
+            .unwrap()
+            .into()
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_empty_raster() -> Result<()> {
+        let no_data_value_option = Some(super::OUT_NO_DATA_VALUE);
+
+        let input = make_empty_raster();
+
+        let op = Radiance {
+            sources: SingleRasterSource { raster: input },
+            params: RadianceParams {},
+        }
+        .boxed()
+        .initialize(&MockExecutionContext::default())
+        .await
+        .unwrap();
+
+        let processor = op.query_processor().unwrap().get_f32().unwrap();
+
+        let ctx = MockQueryContext::new(1);
+        let result_stream = processor
+            .query(
+                RasterQueryRectangle {
+                    spatial_bounds: SpatialPartition2D::new_unchecked(
+                        (0., 4.).into(),
+                        (3., 0.).into(),
+                    ),
+                    time_interval: Default::default(),
+                    spatial_resolution: SpatialResolution::one(),
+                },
+                &ctx,
+            )
+            .await
+            .unwrap();
+
+        let result: Vec<Result<RasterTile2D<PixelOut>>> = result_stream.collect().await;
+
+        assert_eq!(
+            result[0].as_ref().unwrap().grid_array,
+            EmptyGrid2D::new([3, 2].into(), no_data_value_option.unwrap(),).into()
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_missing_offset() -> std::result::Result<(), &'static str> {
+        let input = make_raster(None, Some(2.0));
+
+        let op = Radiance {
+            sources: SingleRasterSource { raster: input },
+            params: RadianceParams {},
+        }
+        .boxed()
+        .initialize(&MockExecutionContext::default())
+        .await
+        .unwrap();
+
+        let processor = op.query_processor().unwrap().get_f32().unwrap();
+
+        let ctx = MockQueryContext::new(1);
+        let result_stream = processor
+            .query(
+                RasterQueryRectangle {
+                    spatial_bounds: SpatialPartition2D::new_unchecked(
+                        (0., 4.).into(),
+                        (3., 0.).into(),
+                    ),
+                    time_interval: Default::default(),
+                    spatial_resolution: SpatialResolution::one(),
+                },
+                &ctx,
+            )
+            .await
+            .unwrap();
+
+        let result: Vec<Result<RasterTile2D<PixelOut>>> = result_stream.collect().await;
+
+        assert_eq!(1, result.len());
+        match &result[0] {
+            Err(_) => Ok(()),
+            Ok(_) => Err("Should fail on missing \"msg.CalibrationOffset\" property."),
+        }
```

Comment:
I guess this is shorter
```suggestion
        assert!(&result[0].is_err());
```

Since our errors are not comparable, you probably can't match on an error.
If you nevertheless want to check for the correct type, you could - in principle - raise a panic and check that (https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#testing-panics).
I'm not sure what version is best.

Reply:
Fixed in: 3be3fd383500c6069bb49ffde1523b83a9c4b184